core/domain/feature_flag_registry.py

```diff
@@ -0,0 +1,176 @@
+# coding: utf-8
+#
+# Copyright 2023 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Registry for feature flags."""
+
+from __future__ import annotations
+
+from core import platform_feature_list
+from core.domain import feature_flag_domain
+from core.platform import models
+
+from typing import List, Optional
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import config_models
+
+(config_models,) = models.Registry.import_models([models.Names.CONFIG])
+
+FeatureNames = platform_feature_list.FeatureNames
+FEATURE_FLAG_NAME_TO_DESCRIPTION_AND_FEATURE_STAGE = (
+    platform_feature_list.FEATURE_FLAG_NAME_TO_DESCRIPTION_AND_FEATURE_STAGE)
+
+
+class Registry:
+    """Registry for all feature flags."""
+
+    @classmethod
+    def get_feature_flag(cls, name: str) -> feature_flag_domain.FeatureFlag:
+        """Returns the instance of the specified name of the feature flag.
+
+        Args:
+            name: str. The name of the feature flag.
+
+        Returns:
+            FeatureFlag. The instance of the specified feature flag.
+
+        Raises:
+            Exception. The given name of the feature flag doesn't exist.
+        """
+        feature_flag_config_from_storage = (
+            cls.load_feature_flag_config_from_storage(name))
+
+        if feature_flag_config_from_storage is not None:
+            feature_flag_spec_values = (
+                FEATURE_FLAG_NAME_TO_DESCRIPTION_AND_FEATURE_STAGE.get(name))
+            assert feature_flag_spec_values is not None
+            feature_flag_spec = feature_flag_domain.FeatureFlagSpec(
+                feature_flag_spec_values[0],
+                feature_flag_spec_values[1]
+            )
+            return feature_flag_domain.FeatureFlag(
+                name,
+                feature_flag_spec,
+                feature_flag_config_from_storage
+            )
+        elif name in FEATURE_FLAG_NAME_TO_DESCRIPTION_AND_FEATURE_STAGE:
+            feature_flag_spec_values = (
+                FEATURE_FLAG_NAME_TO_DESCRIPTION_AND_FEATURE_STAGE.get(name))
+            assert feature_flag_spec_values is not None
+            feature_flag_spec = feature_flag_domain.FeatureFlagSpec(
```

Comment:
Could you please remove the duplication between if/else -- pull out the declaration of feature_flag_spec and also the return value, and only branch when determining the config?